xbmc/music/MusicDatabase.cpp

```diff
@@ -8888,6 +8942,184 @@ bool CMusicDatabase::GetGenresJSON(CFileItemList& items, bool bSources)
+{
+  CMusicDbUrl musicUrl;
+
+  if (!musicUrl.FromString(strBaseDir))
+    return false;
+
+  musicUrl.AddOption("boxset", true);
+
+  Filter filter;
+  return GetAlbumsByWhere(musicUrl.ToString(), filter, items);
+}
+
+bool CMusicDatabase::GetBoxsetDiscs(const std::string& strBaseDir, CFileItemList& items, int idAlbum, const SortDescription &sortDescription)
+{
+  // Need to get the single album - must only be one album holding a boxset
+  // then split the disc names from strDiscTitles, count how many there are
+  // and loop around that many times to create a view with 'i' number of boxes,
+  // titled by the subtitles
+
+  try
+  {
+    if (NULL == m_pDB.get()) return false;
+    if (NULL == m_pDS.get()) return false;
+    std::string strSQL;
+    strSQL=PrepareSQL("SELECT DISTINCT strDiscSubtitle FROM song where song.idAlbum = %i ORDER BY iTrack", idAlbum);
+    // run query
+    if (!m_pDS->query(strSQL)) return false;
+    int iRowsFound = m_pDS->num_rows();
+    if (iRowsFound < 2) // can't only be one disc title for a set
+    {
+      m_pDS->close();
+      return -1;
+    }
+    std::vector<std::string> titlevector;
+     while (!m_pDS->eof())
+    {
+      titlevector.push_back(m_pDS->fv("strDiscSubtitle").get_asString().c_str());
+      m_pDS->next();
+    }
+    m_pDS->close();
+    items.SetProperty("total", iRowsFound);
+
+    CMusicDbUrl musicUrl;
+    std::string temp_path;
+    temp_path = "musicdb://boxsets/";
+    musicUrl.FromString(strBaseDir);
+    musicUrl.RemoveOption("boxset");
+    temp_path += StringUtils::Format("%i/", idAlbum);
+
+    strSQL=PrepareSQL("SELECT albumview.* FROM albumview WHERE albumview.idAlbum = %i", idAlbum);
+    if (!m_pDS->query(strSQL)) return false;
+    int iAlbumsFound = m_pDS->num_rows();
+    if (iAlbumsFound != 1) // must only be one album
+    {
+      m_pDS->close();
+      return -1;
+    }
+    DatabaseResults results;
+    results.reserve(iAlbumsFound);
+    if (!SortUtils::SortFromDataset(sortDescription, MediaTypeSong, m_pDS, results))
+      return false;
+    const dbiplus::sql_record* const record = m_pDS->get_sql_record();
+    CMusicDbUrl itemUrl;
+    new CFileItem(itemUrl.ToString(), GetAlbumFromDataset(record));
+
+    for (const auto &disctitle : titlevector) // loop to construct the discs for the gui
+    {
+      CFileItemPtr pItem(new CFileItem(itemUrl.ToString() , GetAlbumFromDataset(record)));
+      std::string path;
+      path = temp_path;
+      path += StringUtils::Format("%s/", disctitle.c_str());
+      pItem->SetPath(path);
+      pItem->SetLabel(disctitle.c_str());
+      pItem->SetArt("icon", "DefaultAlbumCover.png");
+      items.Add(pItem);
+    }
+  }
+  catch (...)
+  {
+    m_pDS->close();
+    CLog::Log(LOGERROR, "%s failed", __FUNCTION__);
+  }
+  m_pDS->close();
+  return true;
+}
+
+bool CMusicDatabase::GetBoxsetDiscSongs(const std::string& strBaseDir, CFileItemList& items)
+{
```

Comment:
No need for this new method, with its bespoke path parsing and only fetching song data and not the artist credits etc. 
Fetch the song node data using GetSongsByWhere with the correct where clause being built from parameters/options by `GetFilter`.